mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -54,6 +54,17 @@ def __init__(self, artifact_uri):
                                   error_code=INVALID_PARAMETER_VALUE)
         self.run_id = self._extract_run_id(self.artifact_uri)
 
+        # Fetch the artifact root for the MLflow Run associated with `artifact_uri` and compute
+        # the path of `artifact_uri` relative to the MLflow Run's artifact root
+        # (the `run_relative_artifact_repo_root_path`). All operations performed on this artifact
+        # repository will be performed relative to this computed location
+        artifact_repo_root_path = extract_and_normalize_path(artifact_uri)
+        run_artifact_root_uri = self._get_run_artifact_root(self.run_id)
```

Comment:
We fetch the actual run artifact root from the MLflow Tracking Service because it seems somewhat brittle to assume that the root is exactly `dbfs:/databricks/mlflow-tracking/<EXP_ID>/<RUN_ID>/artifacts`. We must assume that it's at least `dbfs:/databricks/mlflow-tracking/<EXP_ID>/<RUN_ID>`, but we don't need to assume that the root contains the `artifacts` subdirectory or that the root is not in some other subdirectory (e.g., `dbfs:/databricks/mlflow-tracking/<EXP_ID>/<RUN_ID>/my/other/awesome/root`).